### PR TITLE
feat(toolbar): Add theme color mixins

### LINKED ISSUE #1154: Add Sass color mixin to toolbar
- [x] Create color mixins
- [x] Call color mixin from stylesheet, with primary color
- [x] Delete any —primary --accent modifiers
- [x] Update README to document color mixins
- [x] Include "breaking change" message about removing modifiers, point to color mixin as deprecation replacement.

### PR DESCRIPTION
**Pull request by acdvorak**
This PR also makes the demo page a little prettier.

Fixes #1154

### CODE REVIEW

File: demos/common.scss [hunk 1]
```diff
@@ -34,20 +35,19 @@ $mdc-theme-secondary: $material-color-light-green-a700 !default;
 @import "../packages/mdc-toolbar/mdc-toolbar";
 @import "../packages/mdc-typography/mdc-typography";
 
+$dark-button-color: $material-color-light-green-a200;
```
Comment (moog16) [inline demos/common.scss]: is this for dark-theme? Is this change needed?

Comment (moog16) [reply demos/common.scss]: Nevermind - I see you just moved it up

File: demos/toolbar/default-flexible-toolbar.html [hunk 2]
```diff
@@ -25,36 +25,11 @@
...
-        color: red;
-        background-color: rgba(0, 0, 0, 0.8);
-      }
     </style>
   </head>
-  <body class="mdc-typography mdc-toolbar-demo">
-    <header class="mdc-toolbar mdc-toolbar--flexible mdc-toolbar--flexible-default-behavior">
-      <div class="mdc-toolbar__row">
+  <body class="mdc-typography">
+    <header class="mdc-toolbar mdc-toolbar--flexible mdc-toolbar--flexible-default-behavior demo-toolbar">
```
Comment (moog16) [inline demos/toolbar/default-flexible-toolbar.html]: should this have the class demo-toolbar--flexible to be consistent with the custom toolbar?

Comment (moog16) [reply demos/toolbar/default-flexible-toolbar.html]: on second thought...its just the demo so nevermind

File: demos/toolbar/waterfall-flexible-toolbar.html [hunk 3]
```diff
@@ -80,7 +53,7 @@
         </p>
       </div>
     </main>
-    <footer>
+    <footer class="demo-toolbar-floating-footer">
```
Comment (moog16) [inline demos/toolbar/waterfall-flexible-toolbar.html]: I think it would be more BEM-ish to be `demo-toolbar__floating-footer` or `demo-toolbar__footer--floating` if we are trying to be BEM

File: demos/toolbar/waterfall-flexible-toolbar-custom-style.html [hunk 4]
```diff
@@ -80,7 +60,7 @@
         </p>
       </div>
     </main>
-    <footer>
+    <footer class="demo-toolbar-floating-footer">
```
Comment (moog16) [inline demos/toolbar/waterfall-flexible-toolbar-custom-style.html]: `demo-toolbar__floating-footer` or `demo-toolbar__footer--floating` comment from below

File: demos/toolbar/waterfall-toolbar-fix-last-row.html [hunk 5]
```diff
@@ -88,7 +63,7 @@
         </p>
       </div>
     </main>
-    <footer>
+    <footer class="demo-toolbar-floating-footer">
```
Comment (moog16) [inline demos/toolbar/waterfall-toolbar-fix-last-row.html]: change here too if we decide to go BEM

File: packages/mdc-toolbar/_mixins.scss [hunk 6]
```diff
@@ -14,14 +14,38 @@
...
+
+@mixin mdc-toolbar-ink-color($color) {
+  @include mdc-theme-prop(color, $color);
+}
+
+@mixin mdc-toolbar-fill-color($color) {
+  @include mdc-theme-prop(background-color, $color);
+}
+
+@mixin mdc-toolbar-fill-color-accessible($color) {
```
Comment (moog16) [inline packages/mdc-toolbar/_mixins.scss]: Confused what it means to be accessible regarding color. Is this like high contrast mode? If so, that seems to me like another theme (like dark-theme).

File: demos/toolbar/waterfall-flexible-toolbar-custom-style.html (hunk 4 above)
Comment (acdvorak) [reply demos/toolbar/waterfall-flexible-toolbar-custom-style.html]: If it were a child of `demo-toolbar`, yes, absolutely! In this particular case, though, it's a sibling of `demo-toolbar`, so it gets its own separate block class.

File: demos/toolbar/waterfall-flexible-toolbar.html (hunk 3 above)
Comment (acdvorak) [reply demos/toolbar/waterfall-flexible-toolbar.html]: Ditto above

File: demos/toolbar/waterfall-toolbar-fix-last-row.html (hunk 5 above)
Comment (acdvorak) [reply demos/toolbar/waterfall-toolbar-fix-last-row.html]: Ditto above

File: packages/mdc-toolbar/_mixins.scss (hunk 6 above)
Comment (acdvorak) [reply packages/mdc-toolbar/_mixins.scss]: Kind of, but not quite. It's not a separate "mode", per se, but simply a convenience mixin for clients who want their button's fill (bg) color to be X, but don't care what the ink (text) color is as long as it's high enough contrast to pass accessibility requirements
([4.5:1 contrast ratio](https://webaim.org/resources/contrastchecker/) or higher).

Assuming that the client has accessible values for `primary` in `$mdc-theme-text-colors`, this mixin guarantees that the ink color it chooses will be accessible on the given fill color.

Comment (moog16) [reply packages/mdc-toolbar/_mixins.scss]: That makes sense...should I also be doing this for drawer?

Comment (acdvorak) [reply packages/mdc-toolbar/_mixins.scss]: Yes - generally speaking, all `-fill-color` mixins should also have a `-fill-color-accessible` version that sets the ink color automatically.
